runtime/lua/vim/filetype.lua

```diff
@@ -2203,12 +2203,12 @@ local pattern = {
     ['/%.cargo/credentials$'] = 'toml',
     ['/%.init/.*%.override$'] = 'upstart',
   },
-  ['calendar/'] = {
+  ['/%.?calendar/'] = {
```

Comment:
Having any quantifiers in parent patterns showed to be *slightly* less performant than not having them. That was the reason for adding [this rule of thumb](https://github.com/neovim/neovim/blob/c9a2b16c4842b5c61af1f57a66637fbd94efba24/runtime/doc/dev_vimpatch.txt#L345-L346).

If adding quantifier visibly reduces chances of false positive parent pattern match, then it is worth adding. I.e. if paths like '/xxx/my-calendar/hello' (which match `'calendar/'` but don't match `'/%.calendar'`) are common. I don't think this is a case here, but that is somewhat subjective.